app/src/main/java/com/babylon/wallet/android/presentation/status/address/AddressDetailsDialog.kt

```diff
@@ -0,0 +1,580 @@
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_copy),
+                    contentDescription = null,
+                    tint = RadixTheme.colors.gray1
+                )
+                Spacer(modifier = Modifier.width(RadixTheme.dimensions.paddingXSmall))
+                Text(text = stringResource(id = R.string.addressDetails_copy))
+            }
+
+            TextButton(
+                onClick = onEnlarge
+            ) {
+                Icon(
+                    painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_enlarge),
+                    contentDescription = null,
+                    tint = RadixTheme.colors.gray1
+                )
+                Spacer(modifier = Modifier.width(RadixTheme.dimensions.paddingXSmall))
+                Text(text = stringResource(id = R.string.addressDetails_enlarge))
+            }
+
+            TextButton(
+                onClick = onShare
+            ) {
+                Icon(
+                    painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_share),
+                    contentDescription = null,
+                    tint = RadixTheme.colors.gray1
+                )
+                Spacer(modifier = Modifier.width(RadixTheme.dimensions.paddingXSmall))
+                Text(text = stringResource(id = R.string.addressDetails_share))
+            }
+        }
+    }
+}
+
+@Composable
+private fun VisitDashboard(
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit
+) {
+    RadixSecondaryButton(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = RadixTheme.dimensions.paddingDefault),
+        text = stringResource(id = R.string.addressDetails_viewOnDashboard),
+        trailingContent = {
+            Icon(
+                modifier = Modifier.size(12.dp),
+                painter = painterResource(id = R.drawable.ic_external_link),
+                contentDescription = null,
+                tint = RadixTheme.colors.gray1
+            )
+        },
+        onClick = onClick
+    )
+}
+
+@Composable
+private fun VerifyAddressOnLedger(
+    modifier: Modifier = Modifier,
+    isVerifying: Boolean,
+    onClick: () -> Unit
+) {
+    RadixSecondaryButton(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = RadixTheme.dimensions.paddingDefault),
+        text = stringResource(id = R.string.addressDetails_verifyOnLedger),
+        onClick = onClick,
+        isLoading = isVerifying,
+        enabled = !isVerifying
+    )
+}
+
+@Composable
+private fun HandleEvents(
+    events: Flow<AddressDetailsDialogViewModel.Event>,
+    onShowEnlargedAddress: (
+        value: String,
+        numberRanges: List<OpenEndRange<Int>>
+    ) -> Unit,
+    onHideEnlargedAddress: () -> Unit
+) {
+    val context = LocalContext.current
+    LaunchedEffect(key1 = events) {
```

Comment:
Do we want to make `events` the key of this effect? As far as I remember in many places we start the effect once, using `Unit`